.github/workflows/ihsBuild.yml

```diff
@@ -183,8 +188,74 @@ jobs:
                 imageResourceId=$imageResourceId vmAdminId=${{ env.vmAdminId }} vmAdminPwd=${{ env.vmAdminPassword }} \
                 scriptLocation=${{ env.scriptLocation }}
             
-            az group delete -n $rgName --yes
+            az group delete -n $rgName --yes --no-wait
+          done
+      - name: Verify the image with twas-cluster integration-test pipeline
+        run: |
+          # Trigger the workflow run
+          requestData={\"ref\":\"main\",\"inputs\":{\"databaseType\":\"none\",\"ndImageResourceId\":\"${ndImageResourceId}\",\"ihsImageResourceId\":\"${ihsImageResourceId}\"}}
+          curl --verbose -L -X POST \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${accessToken}"\
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            https://api.github.com/repos/${{ env.twasClusterRepo }}/actions/workflows/integration-test.yaml/dispatches \
+            -d $(echo $requestData | jq -c)
+          
+          # Wait until the workflow run starts
+          idAndStatus=$(curl -L \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${accessToken}"\
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            https://api.github.com/repos/${{ env.twasClusterRepo }}/actions/workflows/integration-test.yaml/runs \
+            | jq -r '.workflow_runs[0] | {id, status}')
+          status=$(echo $idAndStatus | jq -r '.status')
+          while [[ $status != queued ]]
+          do
+            echo "Wait until the workflow run starts..."
+            idAndStatus=$(curl -L \
+              -H "Accept: application/vnd.github+json" \
+              -H "Authorization: Bearer ${accessToken}"\
+              -H "X-GitHub-Api-Version: 2022-11-28" \
+              https://api.github.com/repos/${{ env.twasClusterRepo }}/actions/workflows/integration-test.yaml/runs \
+              | jq -r '.workflow_runs[0] | {id, status}')
+            status=$(echo $idAndStatus | jq -r '.status')
+            echo "The latest workflow run is ${idAndStatus}"
+          done
+          workflowRunId=$(echo $idAndStatus | jq '.id')
+
+          # Wait until the workflow run completes
+          status=$(curl -L \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${accessToken}"\
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            https://api.github.com/repos/${{ env.twasClusterRepo }}/actions/runs/$workflowRunId \
+            | jq -r '.status')
+          while [[ $status != completed ]]
+          do
+            sleep 10
+            echo "Wait until the workflow run ${workflowRunId} completes..."
```

Comment:
This seems like a very rapid polling interview for a thing that takes so long to complete. I would think it would be on the order of minutes, not seconds. What do you think, @majguo?

Reply:
I agree. What about `sleep 60`?